server/src/com/thoughtworks/go/config/CachedFileGoConfig.java

```diff
@@ -95,9 +96,12 @@ public void onTimer() {
 
     private synchronized void loadFromDisk() {
```

Comment:
I wonder if so many methods in this class need to be synchronized now, given that synchronization is done using a common object.